Objects/stringlib/fastsearch.h

```diff
@@ -48,25 +54,35 @@ STRINGLIB(find_char)(const STRINGLIB_CHAR* s, Py_ssize_t n, STRINGLIB_CHAR ch)
 #else
         /* use memchr if we can choose a needle without two many likely
            false positives */
+        const STRINGLIB_CHAR *s1, *e1;
         unsigned char needle = ch & 0xff;
         /* If looking for a multiple of 256, we'd have too
            many false positives looking for the '\0' byte in UCS2
            and UCS4 representations. */
-        if (needle != 0) {
-            while (p < e) {
+        if (needle != 0)
+            while (e - p > MEMCHR_CUT_OFF) {
```

Comment:
Maybe `do ... while ...` is better here? The first time `e - p > MEMCHR_CUT_OFF` is always true.